### DISCUSSION TITLE
What does path in the workflow refer to

### DISCUSSION THREAD

**Discussion by msuzoagu** [dflook/terraform-github-actions#100] [Q&A] [answered]
This is probably a silly question but what does the `path` input in the snippet refer to? 

> **Inputs**
> * `path`
> 
>   Path to the terraform configuration
> 
>   - Type: string
>   - Optional
>   - Default: The action workspace
> 

I think it means the path to the `main.tf` file but I am not sure.

Also does `default: The action workspace` refer to the default workspace the terraform runs in (when you don't create/select a non-default workspace)?

**Comment by dflook** [comment, marked as answer]
While a GitHub Actions job is running it gets its own directory on disk to use, called it's workspace.
`path` is the relative path to the directory containing your terraform files, from the workspace.

You would almost always use the checkout action to clone your repo into the workspace as the first step of your job, so you can think of `path` as the path to your terraform directory inside your repo. The default being the root of your repo.

Confusingly, Terraform has its own idea of a workspace, which you set with the `workspace` input.
